### PR TITLE
Update sample source guide

### PR DESCRIPTION
**Pull request by aliok**
Fixes #issue-number or description of the problem the PR solves

## Proposed Changes 

- Validate the sample source guide
- Fix the references, file names, etc.
- Adaptions for SinkBinding changes https://github.com/knative/sample-source/pull/40
- Create a separate page for instructions on how to make the source part of `eventing-contrib`

### CODE REVIEW

File: docs/eventing/samples/writing-receive-adapter-source/01-theory.md [hunk 1]
```diff
@@ -83,22 +83,21 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
...
-and
-```patch
-  -i knative.dev/eventing-contrib/camel/source/pkg/apis \
-- -i knative.dev/eventing-contrib/github/pkg/apis
-+ -i knative.dev/eventing-contrib/github/pkg/apis \
-+ -i knative.dev/eventing-contrib/YourSourceHere/pkg/apis
 
+Sample source's [`update-codegen.sh`](https://github.com/knative/sample-source/blob/master/hack/update-codegen.sh) have the configuration
+to have the required things above generated and injected:
+```bash
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/01-theory.md:89]: Remove trailing whitespace:
```suggestion
```bash
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 2]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+1. Update the `ObservedGeneration` and initialize the `Status` conditions (as defined in `samplesource_lifecycle.go` and `samplesource_types.go`)
 ```go
-// Get the resource with this namespace/name.
-original, err := r.Lister.SampleSources(namespace).Get(name)
+src.Status.InitializeConditions()
+src.Status.ObservedGeneration = src.Generation
+```
+2. Create/reconcile the Receive Adapter (detailed below)
+3. If successful, update the `Status` and `MarkDeployed`
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:17]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 3]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+src.Status.ObservedGeneration = src.Generation
+```
+2. Create/reconcile the Receive Adapter (detailed below)
+3. If successful, update the `Status` and `MarkDeployed`
+```go 
+src.Status.PropagateDeploymentAvailability(ra)
+```
+4. Create/reconcile the `SinkBinding` for the Receive Adapter targeting the `Sink` (detailed below)
+5. MarkSink with the result
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:22]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 4]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+```go 
+src.Status.PropagateDeploymentAvailability(ra)
+```
+4. Create/reconcile the `SinkBinding` for the Receive Adapter targeting the `Sink` (detailed below)
+5. MarkSink with the result
+```go 
+src.Status.MarkSink(sb.Status.SinkURI)
+```
+6. Return a new reconciler event stating that the process is done
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:26]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 5]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
 NB The exact arguments may change based on functional requirements
 Create the underlying deployment from the arguments provided, matching pod templates, labels, owner references, etc as needed to fill out the deployment
-Example: [pkg/reconciler/resources/receive_adapter.go](https://github.com/knative/sample-source/tree/master/pkg/reconciler/resources/receive_adapter.go)
+Example: [pkg/reconciler/sample/resources/receive_adapter.go](https://github.com/knative/sample-source/blob/master/pkg/reconciler/sample/resources/receive_adapter.go)
 
 1. Fetch the existing receive adapter deployment
 ```go
-	ra, err := r.KubeClientSet.AppsV1().Deployments(src.Namespace).Get(expected.Name, metav1.GetOptions{})
+namespace := owner.GetObjectMeta().GetNamespace()
+ra, err := r.KubeClientSet.AppsV1().Deployments(namespace).Get(expected.Name, metav1.GetOptions{})
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:53]: Remove trailing whitespace:
```suggestion
ra, err := r.KubeClientSet.AppsV1().Deployments(namespace).Get(expected.Name, metav1.GetOptions{})
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 6]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+```go
+tracker.Reference{
+    APIVersion: appsv1.SchemeGroupVersion.String(),
+    Kind:       "Deployment",
+    Namespace:  ra.Namespace,
+    Name:       ra.Name,
+}
 ```
+2. Fetch the existing `SinkBinding`
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:87]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 7]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+    Name:       ra.Name,
+}
 ```
+2. Fetch the existing `SinkBinding`
+```go 
+namespace := owner.GetObjectMeta().GetNamespace()
+sb, err := r.EventingClientSet.SourcesV1alpha2().SinkBindings(namespace).Get(expected.Name, metav1.GetOptions{})
+```
+2. If it doesn't exist, create it
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:92]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 8]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+namespace := owner.GetObjectMeta().GetNamespace()
+sb, err := r.EventingClientSet.SourcesV1alpha2().SinkBindings(namespace).Get(expected.Name, metav1.GetOptions{})
+```
+2. If it doesn't exist, create it
+```go 
+sb, err = r.EventingClientSet.SourcesV1alpha2().SinkBindings(namespace).Create(expected)
 
+```
+3. Check if the expected vs existing spec is different, and update the `SinkBinding` if required
+```go
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:97]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md [hunk 9]
```diff
@@ -7,64 +7,101 @@ type: "docs"
...
+else if r.specChanged(sb.Spec, expected.Spec) {
+    sb.Spec = expected.Spec
+    if sb, err = r.EventingClientSet.SourcesV1alpha2().SinkBindings(namespace).Update(sb); err != nil {
+        return sb, err
+    }
+```
+4. If updated, record the event
+```go
+return pkgreconciler.NewEvent(corev1.EventTypeNormal, "SinkBindingUpdated", "updated SinkBinding: \"%s/%s\"", namespace, name)
+```
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:107]: Remove trailing whitespace:
```suggestion
```
```

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md [hunk 10]
```diff
@@ -23,10 +22,12 @@ func main() {
 
 ```
 
-## Defining NewAdapter implmentation and Start function
-The adapter's `pkg` implemenation constists of two main functions;
+## Defining NewAdapter implementation and Start function
+The adapter's `pkg` implementation consists of two main functions;
 
-a `NewAdapter(ctx context.Context, aEnv adapter.EnvConfigAccessor, sink cloudevents.Client, reporter source.StatsReporter) adapter.Adapter {}` call.  Which creates the new adapter with passed variables via the EnvConfigAccessor, and sets up the cloudevents client (which is where the events are forwarded to).  This is sometimes refered to as a sink, or ceClient in the knative ecosystem.  The return value is a refernce to the adapter as defined by the adapter's local struct.
+1. A `NewAdapter(ctx context.Context, aEnv adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {}` call, which creates the
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:28]: Remove trailing whitespace:
```suggestion
1. A `NewAdapter(ctx context.Context, aEnv adapter.EnvConfigAccessor, ceClient cloudevents.Client) adapter.Adapter {}` call, which creates the
```

File: docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md [hunk 11]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Adding the event source to eventing-contrib"
+linkTitle: "Adding to eventing-contrib"
+weight: 10
+type: "docs"
+---
+
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md:8]: Remove trailing whitespace:
```suggestion
If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can
```

File: docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Adding the event source to eventing-contrib"
+linkTitle: "Adding to eventing-contrib"
+weight: 10
+type: "docs"
+---
+
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can 
+have a look at different sources there, such as
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md:9]: Remove trailing whitespace:
```suggestion
have a look at different sources there, such as
```

File: docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Adding the event source to eventing-contrib"
+linkTitle: "Adding to eventing-contrib"
+weight: 10
+type: "docs"
+---
+
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can 
+have a look at different sources there, such as  
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source),
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md:10]: Remove trailing whitespace:
```suggestion
[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source),
```

File: docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+title: "Adding the event source to eventing-contrib"
+linkTitle: "Adding to eventing-contrib"
+weight: 10
+type: "docs"
+---
+
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can 
+have a look at different sources there, such as  
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source), 
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md:11]: Remove trailing whitespace:
```suggestion
[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and
```

File: docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+type: "docs"
+---
+
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/), as a starting point you can 
+have a look at different sources there, such as  
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source), 
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and 
+[`AWSSQSSource`](https://github.com/knative/eventing-contrib/tree/master/awssqs).
+
+To generate and inject `clientset`, `cache`, `informers`, and `listers`, ensure that the specific source subdirectories has been added to the injection portion of the
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/07-eventing-contrib.md:14]: Remove trailing whitespace:
```suggestion
To generate and inject `clientset`, `cache`, `informers`, and `listers`, ensure that the specific source subdirectories has been added to the injection portion of the
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 16]
```diff
@@ -1,20 +1,40 @@
...
 
 After completing the tutorial, you'll have a basic event source controller as
 well as receive adapter, which events can be viewed through a basic
-event_display Knative Service.
+`event_display` Knative Service.
 
 Just want to see the code? The reference project is
 [https://github.com/knative/sample-source](https://github.com/knative/sample-source).
 
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:14]: Remove trailing whitespace:
```suggestion
A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 17]
```diff
@@ -1,20 +1,40 @@
...
 After completing the tutorial, you'll have a basic event source controller as
 well as receive adapter, which events can be viewed through a basic
-event_display Knative Service.
+`event_display` Knative Service.
 
 Just want to see the code? The reference project is
 [https://github.com/knative/sample-source](https://github.com/knative/sample-source).
 
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as 
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source),
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:15]: Remove trailing whitespace:
```suggestion
[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source),
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 18]
```diff
@@ -1,20 +1,40 @@
...
 well as receive adapter, which events can be viewed through a basic
-event_display Knative Service.
+`event_display` Knative Service.
 
 Just want to see the code? The reference project is
 [https://github.com/knative/sample-source](https://github.com/knative/sample-source).
 
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as 
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source), 
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:16]: Remove trailing whitespace:
```suggestion
[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 19]
```diff
@@ -1,20 +1,40 @@
...
-event_display Knative Service.
+`event_display` Knative Service.
 
 Just want to see the code? The reference project is
 [https://github.com/knative/sample-source](https://github.com/knative/sample-source).
 
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as 
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source), 
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and 
+[`AWSSQSSource`](https://github.com/knative/eventing-contrib/tree/master/awssqs) that can be used as a reference.
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:17]: Remove trailing whitespace:
```suggestion
[`AWSSQSSource`](https://github.com/knative/eventing-contrib/tree/master/awssqs) that can be used as a reference.
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 20]
```diff
@@ -1,20 +1,40 @@
...
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as 
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source), 
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and 
+[`AWSSQSSource`](https://github.com/knative/eventing-contrib/tree/master/awssqs) that can be used as a reference.  
+
+## Other ways
+
+With the approach in this tutorial, you will create a CRD and a controller for the event source which makes it reusable.
+
+You can also write your own event source using a [ContainerSource](../../../eventing/sources/README.md#meta-sources) which
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:23]: Remove trailing whitespace:
```suggestion
You can also write your own event source using a [ContainerSource](../../../eventing/sources/README.md#meta-sources) which
```

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 21]
```diff
@@ -1,20 +1,40 @@
...
+which provides a framework for injecting environment variables into any Kubernetes resource which has a `spec.template` that looks like a Pod (aka PodSpecable).
+
+
 ## Target Audience
 
 The target audience is already familiar with Kubernetes and Go development and
 wants to develop a new event source, importing their custom events via Knative
 Eventing into the Knative system.
 
+This tutorial is for having your event source in an independent repository.
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/README.md:34]: Remove trailing whitespace:
```suggestion
This tutorial is for having your event source in an independent repository.
```

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md [hunk 22]
```diff
@@ -23,10 +22,12 @@ func main() {
...
+new adapter with passed variables via the `EnvConfigAccessor`. The created adapter will be passed the cloudevents client (which is where the events are forwarded to). This is sometimes refered
```
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:29]: Found potential misspelling(s):
```suggestion
new adapter with passed variables via the `EnvConfigAccessor`. The created adapter will be passed the cloudevents client (which is where the events are forwarded to). This is sometimes referred
```

File: docs/eventing/samples/writing-receive-adapter-source/01-theory.md (hunk 1 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/01-theory.md:89]: Remove trailing whitespace:
```suggestion
```bash
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 2 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:17]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 3 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:22]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 4 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:26]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 5 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:53]: Remove trailing whitespace:
```suggestion
ra, err := r.KubeClientSet.AppsV1().Deployments(namespace).Get(expected.Name, metav1.GetOptions{})
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 6 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:87]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 7 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:92]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 8 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:97]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 9 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:107]: Remove trailing whitespace:
```suggestion
```
```

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md (hunk 22 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:29]: Found potential misspelling(s):
```suggestion
new adapter with passed variables via the `EnvConfigAccessor`. The created adapter will be passed the cloudevents client (which is where the events are forwarded to). This is sometimes referred
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 2 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:17]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 3 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:22]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 4 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:26]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 5 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:53]: Remove trailing whitespace:
```suggestion
ra, err := r.KubeClientSet.AppsV1().Deployments(namespace).Get(expected.Name, metav1.GetOptions{})
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 6 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:87]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 7 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:92]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 8 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:97]: Remove trailing whitespace:
```suggestion
```go
```

File: docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md (hunk 9 above)
Comment (mattmoor) [inline docs/eventing/samples/writing-receive-adapter-source/04-reconciler.md:107]: Remove trailing whitespace:
```suggestion
```
```

File: docs/eventing/samples/writing-receive-adapter-source/01-theory.md [hunk 23]
```diff
@@ -83,22 +83,21 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
...
-+API_DIRS_SOURCES=(github/pkg camel/source/pkg kafka/source/pkg awssqs/pkg couchdb/source/pkg prometheus/pkg YourSourceHere/pkg)
--API_DIRS_SOURCES=(github/pkg camel/source/pkg kafka/source/pkg awssqs/pkg couchdb/source/pkg prometheus/pkg)
-
-```
-and
-```patch
-  -i knative.dev/eventing-contrib/camel/source/pkg/apis \
-- -i knative.dev/eventing-contrib/github/pkg/apis
-+ -i knative.dev/eventing-contrib/github/pkg/apis \
-+ -i knative.dev/eventing-contrib/YourSourceHere/pkg/apis
```
Comment (aliok) [inline docs/eventing/samples/writing-receive-adapter-source/01-theory.md:100]: This part is now explained in a separate page: https://github.com/knative/docs/pull/2475/files#diff-7f8a6b25a6b73452947bc5e926d1e6c9R1

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md [hunk 24]
```diff
@@ -35,80 +36,34 @@ type Adapter struct {
...
-## Reconcile/Create The Receive Adapter
-As part of the source reconciliation, we have to create and deploy
-(and update if necessary) the underlying receive adapter.  The two
-client sets used in this process is the `kubeClientSet` for the
-Deployment tracking, and the `EventingClientSet` for the event
-recording.
-
-Verify the specified kubernetes resources are valid, and update the `Status` accordingly
-
-Assemble the ReceiveAdapterArgs
```
Comment (aliok) [inline docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:80]: This section was actually duplicate of https://github.com/knative/docs/pull/2475/files#diff-df6286b13f3371b30a72bf3277b63a96R30

**Comment by aliok** [comment]
/assign @lberk 
Mind reviewing?

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md [hunk 25]
```diff
@@ -6,8 +6,7 @@ type: "docs"
 ---
 
 ## Receive Adapter cmd
-Similar to the controller, we'll need an injection based main.go similar to the controller
-under `cmd/receiver_adapter/main.go`
+Similar to the controller, we'll need an injection based `main.go` similar to the controller under `cmd/receiver_adapter/main.go`
 ```go
 // This Adapter generates events at a regular interval.
 package main
```
Comment (tom24d) [inline docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:12]: Maybe using `knative.dev/eventing/pkg/adapter/v2` is suggested?

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 26]
```diff
@@ -1,20 +1,40 @@
...
+If you would like to contribute Knative's [`eventing-contrib`](https://github.com/knative/eventing-contrib/) and have your source
+there, there are some instructions at [Adding the event source to `eventing-contrib`](./07-eventing-contrib.md).
```
[the remark this answers is not in the document]
Comment (tom24d) [reply docs/eventing/samples/writing-receive-adapter-source/README.md:36]: Thinking for a bit, If things get settle down, maybe new change might be filed then. It is worth to merge😀

**Comment by tom24d** [comment]
> Thoughts as new source contributor like me. This is my first review in my life so sorry if there is any inappropriate. Thanks!

Ok I learn why we say "assign" first... 😓

File: docs/eventing/samples/writing-receive-adapter-source/README.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (aliok) [reply docs/eventing/samples/writing-receive-adapter-source/README.md:36]: Yeah, let's update this sentence once the stuff is divided into separate repos.

File: docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md (hunk 25 above)
Comment (aliok) [reply docs/eventing/samples/writing-receive-adapter-source/05-receive-adapter.md:12]: That's a good point, but I would prefer leaving it for another PR explaining that `adapter/v2`.

File: docs/eventing/samples/writing-receive-adapter-source/README.md [hunk 27]
```diff
@@ -1,20 +1,40 @@
...
-event_display Knative Service.
+`event_display` Knative Service.
 
 Just want to see the code? The reference project is
 [https://github.com/knative/sample-source](https://github.com/knative/sample-source).
 
+A variety of event sources are available in Knative [`eventing-contrib](https://github.com/knative/eventing-contrib/) such as
+[`KafkaSource`](https://github.com/knative/eventing-contrib/tree/master/kafka/source),
+[`GithubSource`](https://github.com/knative/eventing-contrib/tree/master/github) and
+[`AWSSQSSource`](https://github.com/knative/eventing-contrib/tree/master/awssqs) that can be used as a reference.
```
[the remark this answers is not in the document]
Comment (aliok) [reply docs/eventing/samples/writing-receive-adapter-source/README.md]: thanks! good catch

**Comment by aliok** [comment]
@n3wscott, this needs `lgtm` again

**Comment by aliok** [comment]
@evankanderson I committed the suggestions you made. Thanks.

mind reviewing again?

**Comment by lberk** [comment]
Thanks @aliok 
/lgtm
/approve

**Pull request merged**